src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseService.cs

```diff
@@ -434,14 +435,23 @@ public IEnumerable<Guid> GetReferencedReleaseFileVersions(Guid releaseId, params
         private async Task<Either<ActionResult, bool>> ValidateReleaseSlugUniqueToPublication(string slug,
             Guid publicationId, Guid? releaseId = null)
         {
-            if (await _context.Releases.AnyAsync(r => r.Slug == slug && r.PublicationId == publicationId && r.Id != releaseId))
+            var releases = await _context.Releases
+                .Include(r => r.Publication)
+                .Where(r => r.PublicationId == publicationId).ToListAsync();
+             
+            if (releases.Any(release => release.Slug == slug && release.Id != releaseId && IsLatestVersionOfRelease(releases, release.Id)))
```

Comment:
This condition confuses me (and seems inefficient given the second `Any()` call). Given the above comment about filtering out release on `releaseId`, maybe we could simplify this to:

```csharp
if (releases.Any(release => release.Slug == slug && release.PreviousVersionId != releaseId))
```

Reply:
as previous comment.